# Tasks notebook
***

## Task 1 - Square roots (Newton's method)
Newton's formula from https://math.mit.edu/~stevenj/18.335/newton-sqrt.pdf (page 1).

In [1]:
def sqrt (x):
    n = x 
    while (float(x > 0)) :
        
        #Newton's formula    
        root = 0.5 * (n + (x / n))

        #calculate until the difference 
        #between the root and n is less than 0.01
        if abs(root - n) < 0.01:
            break 
        
        #Returns the last calculated number for n
        n = root
    
    return root

In [2]:
#test
sqrt(4)

2.0000000929222947

## Task 2

In [3]:
#import modules
import numpy as np
import scipy.stats as ss

**Null hypotesis:** There is evidence of an association between drink preference and biscuit preference.

In [7]:
#create array
table = np.array([[43, 57], [56, 45]])

#do the statistics
result = ss.chi2_contingency(table, correction=False)

#show pvalue
result.pvalue

0.07762509678333357

With a significance level of 5% and an observed pvalue of ~7.8% we accept the null hypotesis that there is evidence of association between drink and biscuit preferences.